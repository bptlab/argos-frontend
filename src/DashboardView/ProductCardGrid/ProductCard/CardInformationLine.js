import React, { Component } from 'react';
import './CardInformationLine.css';

class CardInformationLine extends Component {
    render() {
        return (<div></div>
        );
    }
}

export default CardInformationLine;