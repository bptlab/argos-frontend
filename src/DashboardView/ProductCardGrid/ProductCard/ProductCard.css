@charset "UTF-8";
@font-face {
  font-family: "Ionicons";
  src: url("../../../assets/fonts/icons/ionicons.eot?v=2.0.0");
  src: url("../../../assets/fonts/icons/ionicons.eot?v=2.0.0#iefix") format("embedded-opentype"), url("../../../assets/fonts/icons/ionicons.ttf?v=2.0.0") format("truetype"), url("../../../assets/fonts/icons/ionicons.woff?v=2.0.0") format("woff"), url("../../../assets/fonts/icons/ionicons.svg?v=2.0.0#Ionicons") format("svg");
  font-weight: normal;
  font-style: normal; }

/* ###################### PRODUCT CARD ###################### */
.ProductCard {
  height: min-content;
  color: white;
  -webkit-box-shadow: 10px 10px 22px -5px rgba(168, 168, 168, 0.65);
  -moz-box-shadow: 10px 10px 22px -5px rgba(168, 168, 168, 0.65);
  box-shadow: 10px 10px 22px -5px rgba(168, 168, 168, 0.65);
  margin: 1vh;
  min-width: 19vw; }

a .ProductCard:hover {
  text-decoration: none; }

.ProductCard_RUNNING {
  background: #9FAB61 !important; }

.ProductCard_WARNING {
  background: #EF9D6E !important; }

.ProductCard_ERROR {
  background: #D33F49 !important; }

/* ############ PRODUCT CARD INFORMATION LINE ############ */
.ProductCardInformationLine {
  list-style-type: none;
  font-style: normal;
  font-weight: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  padding-right: 1vh;
  line-height: 3vh; }

.ProductCardInformationLine:before {
  font-family: 'Ionicons', Arial, sans-serif;
  margin-right: 1.5vh; }

.PCIL_errorQuantity:before {
  content: ""; }

.PCIL_serialNumber:before {
  content: ""; }

.PCIL_date:before {
  content: ""; }

.PCIL_quantity:before {
  content: ""; }

.icon-check:before {
  content: '\0041'; }

.icon-info-with-circle:before {
  content: '\0044'; }

.icon-warning:before {
  content: '\0046'; }
