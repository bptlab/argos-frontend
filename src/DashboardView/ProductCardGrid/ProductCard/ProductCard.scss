@import "ProductCardHeader/ProductCardHeader";

/* ###################### PRODUCT CARD ###################### */
.ProductCard {
    height: min-content;
    color: white;
    -webkit-box-shadow: 10px 10px 22px -5px rgba(168,168,168,0.65);
    -moz-box-shadow: 10px 10px 22px -5px rgba(168,168,168,0.65);
    box-shadow: 10px 10px 22px -5px rgba(168,168,168,0.65);
    margin-bottom: 30px;
    border-radius: 0px;
    //min-width: 19vw;
}

a .ProductCard:hover {
    text-decoration: none;
}

.ProductCard_RUNNING {
    background: $COLOR-success !important;
    
}

.ProductCard_WARNING {
    background: $COLOR-warning !important;
}

.ProductCard_ERROR {
    background: $COLOR-error !important;
}

.ProductCard_UNDEFINED {
    background: $COLOR-undefined !important;
}

/* ############ PRODUCT CARD INFORMATION LINE ############ */

.ProductCardInformationLine {
    list-style-type: none;
    font-style: normal;
    font-weight: normal;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    padding-right: 1vh;
    line-height: 3vh;
    @include media-breakpoint-down(xs) {
        font-size: 12px;
    }
}

.ProductCardInformationLine:before {
    font-family: 'Ionicons', Arial, sans-serif;
    margin-right: 1.5vh;
}

.PCIL_errorQuantity:before{content: $ICON-errorQuantity;}
.PCIL_serialNumber:before{content:$ICON-serialNumber;}
.PCIL_date:before{content:$ICON-date;}
.PCIL_quantity:before{content: $ICON-quantity;}

.icon-check:before{content:'\0041';}
.icon-info-with-circle:before{content:'\0044';}
.icon-warning:before{content:'\0046';}