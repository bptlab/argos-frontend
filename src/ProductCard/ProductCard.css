@font-face {
  font-family: 'Glyphter';
  src: url("../assets/fonts/icons/Glyphter.eot");
  src: url("../assets/fonts/icons/Glyphter.eot?#iefix") format("embedded-opentype"), url("../assets/fonts/icons/Glyphter.woff") format("woff"), url("../assets/fonts/icons/Glyphter.ttf") format("truetype"), url("../assets/fonts/icons/Glyphter.svg#Glyphter") format("svg");
  font-weight: normal;
  font-style: normal; }

/* ###################### PRODUCT CARD ###################### */
.ProductCard {
  height: min-content;
  color: white;
  -webkit-box-shadow: 10px 10px 22px -5px rgba(168, 168, 168, 0.65);
  -moz-box-shadow: 10px 10px 22px -5px rgba(168, 168, 168, 0.65);
  box-shadow: 10px 10px 22px -5px rgba(168, 168, 168, 0.65);
  margin: 1vh; }

.ProductCard_success {
  background: #7CB342 !important; }

.ProductCard_warning {
  background: #FFB300 !important; }

.ProductCard_error {
  background: #F4511E !important; }

/* ############ PRODUCT CARD INFORMATION LINE ############ */
.ProductCardInformationLine {
  list-style-type: none;
  font-style: normal;
  font-weight: normal;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  padding-right: 1vh;
  line-height: 3vh; }

.ProductCardInformationLine:before {
  font-family: 'Glyphter', Arial, sans-serif;
  margin-right: 1.5vh; }

.PCIL_errorQuantity:before {
  content: "B"; }

.PCIL_serialNumber:before {
  content: "C"; }

.PCIL_date:before {
  content: "E"; }

.PCIL_quantity:before {
  content: "!"; }

.icon-check:before {
  content: '\0041'; }

.icon-info-with-circle:before {
  content: '\0044'; }

.icon-warning:before {
  content: '\0046'; }
