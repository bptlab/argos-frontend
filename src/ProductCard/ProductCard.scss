@import '../App/App.scss';
/* ###################### PRODUCT CARD ###################### */
.ProductCard {
    height: min-content;
    color: white;
    -webkit-box-shadow: 10px 10px 22px -5px rgba(168,168,168,0.65);
    -moz-box-shadow: 10px 10px 22px -5px rgba(168,168,168,0.65);
    box-shadow: 10px 10px 22px -5px rgba(168,168,168,0.65);
    margin: 1vh;
}

.ProductCard_success {
    background: $COLOR-success !important;
}

.ProductCard_warning {
    background: $COLOR-warning !important;
}

.ProductCard_error {
    background: $COLOR-error !important;
}

/* ############ PRODUCT CARD INFORMATION LINE ############ */

.ProductCardInformationLine {
    list-style-type: none;
    font-style: normal;
    font-weight: normal;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    padding-right: 1vh;
    line-height: 3vh;
}

.ProductCardInformationLine:before {
    font-family: 'Glyphter', Arial, sans-serif;
    margin-right: 1.5vh;
}

.PCIL_errorQuantity:before{content: $ICON-errorQuantity;}
.PCIL_serialNumber:before{content:$ICON-serialNumber;}
.PCIL_date:before{content:$ICON-date;}
.PCIL_quantity:before{content: $ICON-quantity;}

.icon-check:before{content:'\0041';}
.icon-info-with-circle:before{content:'\0044';}
.icon-warning:before{content:'\0046';}