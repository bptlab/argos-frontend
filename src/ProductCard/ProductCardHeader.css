@font-face {
  font-family: 'Glyphter';
  src: url("../assets/fonts/icons/Glyphter.eot");
  src: url("../assets/fonts/icons/Glyphter.eot?#iefix") format("embedded-opentype"), url("../assets/fonts/icons/Glyphter.woff") format("woff"), url("../assets/fonts/icons/Glyphter.ttf") format("truetype"), url("../assets/fonts/icons/Glyphter.svg#Glyphter") format("svg");
  font-weight: normal;
  font-style: normal; }

.ProductCardTitle {
  width: 100%;
  text-align: center;
  padding: 0.5vw 2vw;
  border-bottom: 1px solid white;
  font-size: 1.6vh; }

.ProductCardTitle_success {
  background: #689F38 !important; }

.ProductCardTitle_warning {
  background: #FFA000 !important; }

.ProductCardTitle_error {
  background: #E64A19 !important; }
