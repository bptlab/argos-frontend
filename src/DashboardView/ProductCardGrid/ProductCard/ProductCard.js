import React, { Component } from 'react';
import {Card} from 'reactstrap';
import './ProductCard.css';
import ProductCardHeader from './ProductCardHeader/CardHeader.js';

class ProductCard extends Component {
    render() {
        return (
            <div className="col-xs-12 col-sm-6 col-md-4 col-lg-3">
                <a href={ `/#/product/${this.props.product.id}`} title={ `Details for ${this.props.product.name}`}>
                    <Card className={ `ProductCard ProductCard_${this.props.product.state}` }>
                        <ProductCardHeader category={this.props.product.state}>
                            {this.props.product.name}
                        </ProductCardHeader>
                        <ul>
                            <ProductCardInformationLine category="serialNumber">
                                {this.props.product.orderNumber}
                            </ProductCardInformationLine>
                            <ProductCardInformationLine category="quantity">
                                {this.props.product.numberOfDevices}
                            </ProductCardInformationLine>
                            <ProductCardInformationLine category="date">
                                {this.props.product.productionStart}
                            </ProductCardInformationLine>
                            <ProductCardInformationLine category="errorQuantity">
                                {this.props.product.numberOfEvents}
                            </ProductCardInformationLine>
                            </ul>
                    </Card>
                </a>
            </div>
        );
    }
}
export default ProductCard;

class ProductCardInformationLine extends Component {
    render() {
        return(
            <li className={ `ProductCardInformationLine PCIL_${this.props.category}` }>
                {this.props.children}
            </li>
        )
    }
}