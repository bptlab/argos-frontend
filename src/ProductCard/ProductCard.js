import React, { Component } from 'react';
import {Card} from 'reactstrap';
import './ProductCard.css';
import ProductCardHeader from './ProductCardHeader.js';

class ProductCard extends Component {
    render() {
        return (
            <Card className={ `ProductCard ProductStatus_${this.props.category}` }>
                <ProductCardHeader category={this.props.category}> Testbrand1 Lgamax Plus GB172-24, EG-E</ProductCardHeader>
                    <ul>
                        <ProductCardInformationLine category="serialNumber">7716 0104 16</ProductCardInformationLine>
                        <ProductCardInformationLine category="quantity">24.156</ProductCardInformationLine>
                        <ProductCardInformationLine category="date">11.11.2016</ProductCardInformationLine>
                        <ProductCardInformationLine category="errorQuantity">4</ProductCardInformationLine>
                    </ul>
            </Card>
        );
    }
}
export default ProductCard;

class ProductCardInformationLine extends Component {
    render() {
        return(<li className={ `ProductCardInformationLine PCIL_${this.props.category}` }>{this.props.children}</li>)
    }
}