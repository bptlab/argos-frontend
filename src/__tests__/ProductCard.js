import React from 'react';
import ProductCard from '../DashboardView/ProductCardGrid/ProductCard/ProductCard.js';
import TestData from './testData/products';
import renderer from 'react-test-renderer';

test('Render DashboardView Header', () => {
    const component = renderer.create(
        <ProductCard product={TestData.PRODUCTS[0]} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});