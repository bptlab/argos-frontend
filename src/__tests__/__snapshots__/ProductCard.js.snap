exports[`test Render DashboardView Header 1`] = `
<div
  className="col-xs-12 col-sm-6 col-md-4 col-lg-3">
  <a
    href="/#/product/0"
    title="Details for Testproduct 5000 14E-17B">
    <div
      className="card ProductCard ProductCard_RUNNING">
      <h4
        className="card-title ProductCardTitle ProductCardTitle_RUNNING">
        Testproduct 5000 14E-17B
      </h4>
      <ul>
        <li
          className="ProductCardInformationLine PCIL_serialNumber">
          9830510416
        </li>
        <li
          className="ProductCardInformationLine PCIL_quantity">
          52189
        </li>
        <li
          className="ProductCardInformationLine PCIL_date">
          2016-05-07
        </li>
        <li
          className="ProductCardInformationLine PCIL_errorQuantity">
          84
        </li>
      </ul>
    </div>
  </a>
</div>
`;
